core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -152,13 +156,19 @@ class KafkaApis(val requestChannel: RequestChannel,
     // ensureTopicExists is only for client facing requests
     // We can't have the ensureTopicExists check here since the controller sends it as an advisory to all brokers so they
     // stop serving data to clients for the topic being deleted
-    val stopReplicaRequest = request.requestObj.asInstanceOf[StopReplicaRequest]
+    val stopReplicaRequest = request.body.asInstanceOf[StopReplicaRequest]
 
-    authorizeClusterAction(request)
```

Comment:
@ijuma I will add a jira to cleanup KafkaApis as part of the parent jira with a comment to be sure to handle this exact case. Its work I was thinking of doing anyway.